src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentMarkupEncodingPass.cs

```diff
@@ -64,7 +68,68 @@ public override void VisitHtml(HtmlContentIntermediateNode node)
                             return;
                         }
                     }
+
+                    token.Content = DecodeHtmlEntities(token.Content);
```

Comment:
It's worth a comment explaining that we only want to decode entities when we are treating the text as encoded.

This code also has a subtle bug. Consider something like:
```html
<div>Ted Cruz &amp; Tom Cruise @* people Ajay wants to go on a cruise with *@ are
cool
</div>
```

This will decompose into one html node with two tokens (I think). 

The pass that decodes entities needs to be a separate loop that runs when `SetEncoded` as not called.

Reply:
> This code also has a subtle bug.

Technically the output will not change just because we decode it even for `AddMarkupContent`. But I get it, we don't have to decode it if we're gonna end up calling AddMarkupContent anyway